web3.js/src/transaction/versioned.ts

```diff
@@ -0,0 +1,109 @@
+import nacl from 'tweetnacl';
+import * as BufferLayout from '@solana/buffer-layout';
+
+import {Signer} from '../keypair';
+import assert from '../utils/assert';
+import {VersionedMessage, deserializeMessage} from '../message/versioned';
+import {SIGNATURE_LENGTH_IN_BYTES} from './constants';
+import * as shortvec from '../utils/shortvec-encoding';
+import * as Layout from '../layout';
+
+export type TransactionVersion = 'legacy' | number;
```

Comment:
What about typing this as `'legacy' | 0` or an enum? Any new version would presumably require a change here anyway.